tools/backsub/backsub.xml

```diff
@@ -0,0 +1,96 @@
+<tool id="backsub" name="Background subtraction" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="@PROFILE@">
+
+    <description>for sequential immunofluorescence images.</description>
+
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+ 
+    <expand macro="requirements"/>
+    <expand macro="version_cmd"/>
+
+    <command detect_errors="aggressive"><![CDATA[
+
+        @CMD_BEGIN@ 
+
+        ##Supply image
+        --root $root
+
+        ##Supply markers
+        --markers $markers
+
+        ##Name output image file
+        --output out.ome.tiff
+
+        ##Name output markers file
+        --marker-output out.csv
+
+        ##Additional arguments
+        #if $adv.pixel_size
+        --pixel-size $adv.pixel_size
+        #end if
+
+        #if $adv.tile_size
+        --tile-size $adv.tile_size
+        #end if
+        
+    ]]></command>
+
+    <inputs>
+        <param name="root" type="data" format="ome.tiff" optional="false" label="Image to process"/>
+        <param name="markers" type="data" format="csv" optional="false" label="Markers file mapping channels to subtract" help="Expected columns: marker_name, background, exposure, remove"/>
+        <section name="adv" title="Advanced Options" expanded="false">
+            <param name="pixel_size" type="float" optional="true" label="Pixel size in microns" help="If not supplied, finds resolution in XML metadata, otherwise defaults to 1"/>
+            <param name="tile_size" type="integer" optional="true" label="Tile size for pyramid generation (default 1024)"/>
+        </section>
+    </inputs>
+
+    <outputs>
+        <data format="ome.tiff" name="image_output" from_work_dir="out.ome.tiff" label="${tool.name} on ${on_string}: Subtracted Image"/>
+        <data format="csv" name="marker_output" from_work_dir="out.csv" label="${tool.name} on ${on_string}: Markers"/>
+    </outputs>
+
+    <tests>
+        <test expect_num_outputs="2">
+            <param name="root" value="test.ome.tiff" />
+            <param name="markers" value="test_markers.csv" />
+            <param name="tile_size" value="256" />
+            <output name="image_output" ftype="ome.tiff">
+                <assert_contents>
+                    <has_size value="520000" delta="10000" />
```

Comment:
Just as information for future udpates, the latest Galaxy release has some nice image assertions: https://docs.galaxyproject.org/en/latest/dev/schema.html#assertions-for-image-data

So we can now design tests that are way more stringent.

Reply:
Thank you very much for the review and feedback @bgruening. Good to know about the new image assertions, I'll create an issue in our main repo so we remember to update that across our tools when we update our instances and CI to the latest Galaxy version